### PR TITLE
remove mco dep and associated functionality

### PR DESCRIPTION
**Pull request by berndbischl**
we need to
- search for nsga2
- search for mco-calls
- updates NEWS

most TODO seems to be that I removed infillOptMultiObjNSGA2 and some opinions remain in control object classes

### CODE REVIEW

File: R/renderExampleRunPlot_helpers.R [hunk 1]
```diff
@@ -50,22 +50,22 @@ getIDX = function(opt.path, i) {
...
-      rep("init", length(idx$init)),
-      rep("seq", length(idx$seq)),
-      rep("prop", length(idx$proposed)),
-      rep("front", length(idx.nsga2.paretofront))
-    ))
-  )
-  names(df) = c(name, "type")
-  return(df)
-}
+# # Get Dataset for Plotting 2D XSpace
```
Comment (jakob-r) [inline R/renderExampleRunPlot_helpers.R:53]: A comment why this is commented out would help for future readers.

File: tests/testthat/test_plot_MBOResult.R [hunk 2]
```diff
@@ -16,16 +16,16 @@ test_that("plot single-objective", {
...
-  f = smoof::makeZDT1Function(2L)
-  learner = makeLearner("regr.km", predict.type = "se")
-  des = generateTestDesign(8L, getParamSet(f))
-  ctrl = makeMBOControl(propose.points = 2L, n.objectives = 2L)
-  ctrl = setMBOControlTermination(ctrl, iters = 2L)
-  ctrl = setMBOControlInfill(ctrl, crit = crit.ei, opt.focussearch.points = 100L,
-    opt.focussearch.maxit = 3L, opt = "nsga2")
-  ctrl = setMBOControlMultiObj(ctrl, method = "mspot")
-  or = mbo(f, des, learner = learner, control = ctrl)
+# test_that("plot multi-objective", {
```
Comment (jakob-r) [inline tests/testthat/test_plot_MBOResult.R:19]: As above. Also mention what would be necessary to insert this again.

**Comment by jakob-r** [comment]
But this cuts some functionality right? So probably the mboControl stuff should be updated, or we are just ok with users getting strange errors if they try to use specific settings?

**Comment by jakobbossek** [comment]
Olaf already [fixed the issues](https://github.com/olafmersmann/mco/commits/master) some days ago. Seems that [not all system checks are finished so far on CRAN](https://cran.r-project.org/web/checks/check_results_mco.html) though.

**Comment by jakob-r** [comment]
mco has been fixed on CRAN so this in not necessary (for now)

**Pull request closed without merging**